### DISCUSSION TITLE
Widget showing alert

### DISCUSSION THREAD

**Discussion by VibeDesign** [pretix/pretix#3765] [Q&A]
I am using the widget to embed the pretix shop into my site. Now, the overall display of the products works fine, but I always have a black exclamation mark in a circle at the bottom of the page (inside the `.pretix-widget-overlay`). Where can I find info what is causing this? I assume this is some sort of error?

Also when I proceed to place a product in the cart, everything shows up inside the mentioned overlay but below the `.pretix-widget-wrapper` instead of inside/on top of it (atleast that's what I expected). Is this normal behaviour or is something wrong?

**Comment by wiffbi** [comment]
Do you have an example URL where we can have a look?

**Reply by VibeDesign** [reply]
You can have a look [here](https://hamburger-singewettstreit.de/ticket-demo). On Desktop View, if you scroll all the way down it also jumps back up, this doesn't happen if you shrink the site to the point, where the left navigation is hidden.

**Reply by wiffbi** [reply]
This is an issue somehow with your sidebar and scroll.sidebar-lock script in you main.js. If the main content is shorter than the sidebar and you overscroll the page, it jumps as described – on any page. So IMHO not an issue with pretix-widget. As soon as I disable your main.js it does not do the jumping anymore.

**Reply by VibeDesign** [reply]
Yes, I assumed something like that. Like I mentioned as soon as you are on a smaller viewport nothing is jumping anymore. But that's not my main issue. Why is there an exclamation mark icon and why is the cart appended at the bottom of the widget instead of replacing the products iframe. Is this an error or is this the intended way (by pretix)?

Because currently I can add something to the cart, then the cart pops up somewhat randomly below, but then when I go back inside the lower iframe, I get the whole product overview again. That all seems poorly built at the moment.

**Reply by wiffbi** [reply]
I think you are missing all default widget CSS styles.

**Reply by VibeDesign** [reply]
Yes, I don't have the default styles included, but I also didn't see anything referring to the div `pretix-widget-overlay` which holds everything except the products. 

But you are confirming to me, that the way it's currently looking is not the intended way? That's all I need to know for now, so I can keep looking for errors.

Also you still didn't mention anything about the exclamation mark icon, any idea why this is showing?

**Reply by wiffbi** [reply]
Yes, it is not the intended way how that is looking. The overlay is used when the checkout process is started – this needs to happen inside an iframe which lives inside an overlay usually centered on the screen. For that to work, you need to load the default widget CSS styles.

**Reply by wiffbi** [reply]
The exclamation mark will be hidden through the default widget styles. AS you do not load the css, it will not be hidden.
